File: result/metagpt/claude/humaneval/134.c
#include <stdbool.h>
#include <string.h>

bool check_if_last_char_is_a_letter(const char* txt) {
    int length = strlen(txt);
    if (length == 0) return false;
    
    char chr = txt[length - 1];
    if (chr < 65 || (chr > 90 && chr < 97) || chr > 122) return false;
    
    if (length == 1) return true;
    
    chr = txt[length - 2];
    if ((chr >= 65 && chr <= 90) || (chr >= 97 && chr <= 122)) return false;
    
    return true;
}

File: result/metagpt/gpt4o/humaneval/134.rs
fn check_if_last_char_is_a_letter(txt: &str) -> bool {
    let length = txt.len();
    if length == 0 {
        return false;
    }

    let chr = txt.chars().nth(length - 1).unwrap();
    if !chr.is_alphabetic() {
        return false;
    }

    if length == 1 {
        return true;
    }

    let prev_chr = txt.chars().nth(length - 2).unwrap();
    if prev_chr.is_alphabetic() {
        return false;
    }

    true
}

fn main() {
    let txt = "Hello!";
    let result = check_if_last_char_is_a_letter(txt);
    println!("Is the last character a letter? {}", result);
}